src/main/java/com/ericsson/ei/services/SubscriptionService.java

```diff
@@ -64,24 +67,40 @@
     @Autowired
     private ISubscriptionRepository subscriptionRepository;
 
+    @Autowired
+    private Encryptor encryptor;
+
     @Override
-    public void addSubscription(Subscription subscription) throws JsonProcessingException, MongoWriteException {
+    public void addSubscription(Subscription subscription)
+            throws JsonProcessingException, MongoWriteException {
+        String authType = subscription.getAuthenticationType();
+        String username = subscription.getUserName();
+        String password = subscription.getPassword();
+        boolean authenticationDetailsProvided = EncryptionFormatter.verifyAuthenticationDetails(
+                authType,
+                username, password);
+        if (authenticationDetailsProvided && encryptor.isJasyptPasswordSet()
```

Comment:
Please extract the content inside () intoo a method with a nice name, like passwordShouldBeEncrypotatoe()

Reply:
Done.